src/pages/Location/Location.js

```diff
@@ -260,6 +344,129 @@ const LocationComponent = () => {
+      });
+    }
+  };
+
+  const handleClose = (save) => {
+    const { actionType, areaType, fieldName } = fieldDialogState;
+    if (save) {
+      if (actionType === 'add') {
+        const errText = fieldNameValidation(fieldName, userFieldsRef.current);
+        if (errText !== '') {
+          setFieldDialogState({ ...fieldDialogState, error: true, errorText: errText });
+          return;
+        }
+        const { longitude, latitude } = selectedToEditSite;
+        const point = buildPoint(longitude, latitude, fieldName);
+        // console.log('add area', polygon);
+        if (areaType === 'Point') {
+          userFieldsRef.current = [...userFieldsRef.current, point];
+        }
+        if (areaType === 'Polygon') {
+          const polygon = currArea.features.slice(-1)[0];
+          const geoCollection = buildGeometryCollection(point.geometry, polygon.geometry, fieldName);
+          userFieldsRef.current = [...userFieldsRef.current, geoCollection];
+        }
+        setSelectedUserField(fieldName);
+      }
+      if (actionType === 'update') {
+        // Only supports polygon updates
+        const { longitude, latitude } = selectedToEditSite;
+        const point = buildPoint(longitude, latitude, selectedUserField);
+        const polygon = currArea.features.slice(-1)[0];
+        // console.log('update', polygon);
+        const geoCollection = buildGeometryCollection(point.geometry, polygon.geometry, selectedUserField);
+        userFieldsRef.current = userFieldsRef.current.map((f) => {
+          if (f.label === selectedUserField) return geoCollection;
+          return f;
+        });
+      }
+      if (actionType === 'delete') {
+        // add a delete flag so when saving the code knows delete which field
+        userFieldsRef.current = userFieldsRef.current.map((f) => {
+          if (f.label === selectedUserField) {
+            // if this field have not been saved to the backend, directly delete it from the userFields.
+            if (!f.id) return null;
+            return { ...f, delete: true };
+          }
+          return f;
+        });
+        userFieldsRef.current = userFieldsRef.current.filter((f) => f !== null);
+        // find next available field in current fields array
+        const nextAvailableField = userFieldsRef.current.find((f) => f.delete !== true);
+        setSelectedUserField(nextAvailableField === undefined ? '' : nextAvailableField.label);
+      }
+      if (actionType === 'updateName') {
+        const { prevName } = fieldDialogState;
+        const errText = fieldNameValidation(fieldName, userFieldsRef.current);
+        if (errText !== '') {
+          setFieldDialogState({ ...fieldDialogState, error: true, errorText: errText });
+          return;
+        }
+        const newField = {
+          type: 'Feature',
+          geometry: userFieldsRef.current.filter((f) => f.label === prevName)[0].geometry,
```

Comment:
all of these maps should have the "f" prop from the map changed to something more readable. i suggest "userFieldRef" or "userField" ill leave comments on the others i found too.